Source/Core/VideoBackends/D3D12/Render.cpp

```diff
@@ -1244,6 +1242,8 @@ void Renderer::SetSamplerState(int stage, int tex_index, bool custom_tex)
 	if (tex_index)
 		stage += 4;
 
+	const auto original_hex = gx_state.sampler[stage].hex;
```

Comment:
@lioncash Okay. I was interested because bitfields are one of those problems that keeps coming up but never seems to have a satisfactory solution. If you're dealing with some binary file format, like an audio file or such, the easiest thing to do is just memory map it then cast the pointer into a pointer to a packed struct to parse it. That's nice and all until you start thinking about portability and trying to only use standard language features.

I suppose the cleanest solution for this sort of problem is to use some sort of [BinaryReader](https://msdn.microsoft.com/en-us/library/system.io.binaryreader(v=vs.110).aspx) then construct bit decomposition objects like your IND_MTXA example from the extracted primitives.

Anyway, thanks for taking the time to walk me through this.